File: frontend/src/api/template.ts
/**
 * 模版 API — 与后端 routers/template.py 对应
 *
 * 端点：
 *   GET  /api/template
 *   POST /api/template/items
 *
 * 主要由 T-FE-C 的 TemplateManager 页面调用，但 T-FE-A 也写模块
 * 是为了在 CONTEXT-04 §2.2 共享 types 一并到位。
 */

import apiClient from './client'
import type { Template, PromoteRequest, PromoteResponse } from '@/types'

export async function getTemplate(): Promise<Template> {
  const { data } = await apiClient.get<Template>('/template')
  return data
}

export async function promoteItem(payload: PromoteRequest): Promise<PromoteResponse> {
  const { data } = await apiClient.post<PromoteResponse>('/template/items', payload)
  return data
}

File: backend/app/routers/template.py
"""模版管理路由。"""
import json
from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy.orm import Session

from app.config import settings
from app.database import get_db
from app.schemas import TemplateResponse, PromoteRequest, PromoteResponse
from app.services.item_service import promote_to_template
from app.services.project_service import get_or_load_template_items

router = APIRouter(prefix="/api/template", tags=["template"])


@router.get("", response_model=TemplateResponse)
def get_template():
    items = get_or_load_template_items()
    with open(settings.TEMPLATE_PATH, "r", encoding="utf-8") as f:
        data = json.load(f)
    return TemplateResponse(version=data.get("version", 1), items=items)


@router.post("/items", response_model=PromoteResponse)
def promote_item(payload: PromoteRequest):
    result = promote_to_template(payload.name, payload.description)
    return PromoteResponse(**result)
